### PR TITLE
EZP-31240: Added ContentTranslationHandler for delete translation document in solr

### PR DESCRIPTION
**Pull request by mateuszdebinski**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31420](https://jira.ez.no/browse/EZP-31420)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `7.5`
| **BC breaks**      | no
| **Tests pass**     | [yes](https://travis-ci.org/github/ezsystems/ezpublish-kernel/builds/675704022)
| **Doc needed**     | no

## Description 
Depends on: https://github.com/ezsystems/ezplatform-solr-search-engine/pull/173

Removing a content translation does not remove the associated SolR documents

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/API/Repository/Tests/SearchDeleteTranslationTest.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+     * @return \eZ\Publish\API\Repository\Values\Content\Content
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentFieldValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    protected function createTestContentByLanguages(array $languages): Content
```
Comment (konradoboza) [inline eZ/Publish/API/Repository/Tests/SearchDeleteTranslationTest.php:51]: Naming, maybe `createTestContentWithLanguages`?

File: eZ/Publish/API/Repository/Tests/SearchDeleteTranslationTest.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+use eZ\Publish\API\Repository\Values\Content\Search\SearchResult;
+
+/**
+ * Test case for field filtering operations in the SearchService.
+ *
+ * @see \eZ\Publish\API\Repository\SearchService
+ * @group integration
+ * @group search
+ */
+class SearchDeleteTranslationTest extends BaseTest
```
Comment (konradoboza) [inline eZ/Publish/API/Repository/Tests/SearchDeleteTranslationTest.php:24]:
```suggestion
class SearchServiceDeleteTranslationTest extends BaseTest
```

File: phpunit-integration-legacy-solr.xml [hunk 3]
```diff
@@ -59,6 +59,7 @@
             <file>eZ/Publish/API/Repository/Tests/SearchServiceLocationTest.php</file>
             <file>eZ/Publish/API/Repository/Tests/SearchServiceAuthorizationTest.php</file>
             <file>eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php</file>
+            <file>eZ/Publish/API/Repository/Tests/SearchDeleteTranslationTest.php</file>
```
Comment (konradoboza) [inline phpunit-integration-legacy-solr.xml:62]:
```suggestion
            <file>eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php</file>
```

**Comment by mateuszdebinski** [comment]
> CS is failing. I need to see passing CI for Solr

File: eZ/Publish/SPI/Search/ContentTranslationHandler.php [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\SPI\Search;
```
Comment (alongosz) [inline eZ/Publish/SPI/Search/ContentTranslationHandler.php:9]:
```suggestion
 */
 declare(strict_types=1);
 
namespace eZ\Publish\SPI\Search;
```

File: eZ/Publish/SPI/Search/ContentTranslationHandler.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+ */
+interface ContentTranslationHandler
+{
+    /**
+     * Deletes a translation content object from the index.
+     *
+     * @param int $contentId
+     * @param string $languageCode
+     */
+    public function deleteTranslation($contentId, $languageCode);
```
Comment (alongosz) [inline eZ/Publish/SPI/Search/ContentTranslationHandler.php:22]: This is PHP 7+ ;)
```suggestion
    /**
     * Delete a translation Content item from the search index.
     */
    public function deleteTranslation(int $contentId, string $languageCode): void;
```

File: eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        return $searchService->findContent($query, $languageFilter);
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentFieldValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php:113]: AFAIR Some of those throw common ForbiddenException. Replace them with that please.

File: eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+    /**
+     * @param array $languages
+     * @return \eZ\Publish\API\Repository\Values\Content\Content
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentFieldValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php:49]: AFAIR Some of those throw common ForbiddenException. Replace them with that please.

File: eZ/Publish/SPI/Search/ContentTranslationHandler.php [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
```
Comment (alongosz) [inline eZ/Publish/SPI/Search/ContentTranslationHandler.php:5]: CS
```suggestion
```

File: eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php [hunk 9]
```diff
@@ -0,0 +1,132 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php:5]: CS
```suggestion
```

File: eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\API\Repository\Tests;
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php:9]:
```suggestion
 */
 declare(strict_types=1);
 
namespace eZ\Publish\API\Repository\Tests;
```

File: eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+ * @see \eZ\Publish\API\Repository\SearchService
+ * @group integration
+ * @group search
+ */
+class SearchServiceDeleteTranslationTest extends BaseTest
+{
+    /**
+     * @throws \ErrorException
+     */
+    public function setUp()
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php:29]: Technical debt:
```suggestion
    public function setUp(): void
```

File: eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+            [$locationCreateStructArticle]
+        );
+        $content = $contentService->publishVersion($draftArticle->getVersionInfo());
+        $this->refreshSearch($repository);
+
+        return $content;
+    }
+
+    /**
+     * @return \eZ\Publish\API\Repository\Values\Content\Search\SearchResult
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php:88]: CS, redundant
```suggestion
```

File: eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentFieldValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ContentValidationException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     */
+    public function testDeleteContentTranslation()
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php:115]:
```suggestion
    public function testDeleteContentTranslation(): void
```

File: eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php [hunk 14]
```diff
@@ -0,0 +1,132 @@
...
+use eZ\Publish\API\Repository\Values\Content\Search\SearchResult;
+
+/**
+ * Test case for delete content translation with the SearchService
+ *
+ * @see \eZ\Publish\API\Repository\SearchService
+ * @group integration
+ * @group search
+ */
+class SearchServiceDeleteTranslationTest extends BaseTest
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/SearchServiceDeleteTranslationTest.php:24]: There's a lot of those already. Don't we have `SearchService`-is namespace already here or in `master`? Not asking to move everything, just this one (if indeed we have ns)

**Comment by mateuszdebinski** [comment]
Tests: https://travis-ci.org/github/ezsystems/ezpublish-kernel/builds/675704022

**Comment by mateuszdebinski** [comment]
PR for merge up: https://github.com/ezsystems/ezplatform-kernel/pull/58

**Comment by lserwatka** [comment]
You can merge it up.

**Pull request merged**